tests/test_directives.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import pytest
+import sys
+from copy import deepcopy
+
+from flow.directives import Directives
+from flow.directives import NP, NRANKS, NGPU, EXECUTABLE, OMP_NUM_THREADS
+
+
+@pytest.fixture()
+def setUp(test_class=Directives, **kwargs):
+    def _setup(test_class=Directives, **kwargs):
+        return test_class(**kwargs)
+    return _setup
+
+
+available_directives_list = [NP, NRANKS, NGPU, EXECUTABLE, OMP_NUM_THREADS]
+
+
+class TestItems:
+    """
+    Tests for _DirectivesItem class
+    """
+    def test_serial(self):
+        assert NP.serial(4, 2) == 4
+        assert not NP.serial('4', '2') == 4
+
+    def test_parallel(self):
+        assert NP.parallel(4, 2) == 6
+
+    def test_parallel_manual(self):
+        directive_copy = deepcopy(NP)
+        directive_copy.parallel = lambda x, y: x*y/(x+y)
+        assert directive_copy.parallel(3, 2) == 1.2
+        with pytest.raises(TypeError):
+            directive_copy.parallel('4', '2')
+        with pytest.raises(ZeroDivisionError):
+            directive_copy.parallel(0, 0)
+
+    def test_validation(self):
+        for directive in available_directives_list:
+            directive.validation(directive.default)
```

Comment:
This tests that the default is a valid value. This may not be the case (if the default is `None`).

Reply:
Checked for `None` type

Reply:
I do not see the change. Did you forget to push your changes?

Reply:
Anyways for this test, I would prefer to see not that the default works, but that the validation functions work (i.e. that given appropriate values they set and given inappropriate values they error).

Reply:
About your latest comment, I think this should be addressed internally itself rather than manually. What I mean to say is, while setting a default value, we should check whether the validation returns a proper value or not.

Reply:
I am not sure as to why we should test that behavior though? If you are wanting to test that the defaults are valid to set to the directive, then we should name this `test_defaults_are_valid` or something similar.